### DISCUSSION TITLE
DBeaver connected to DB2 spawn slow/long running tasks when using Database Navigator

### DISCUSSION THREAD

**Discussion by robertwx** [dbeaver/dbeaver#15343] [Q&A]
Problem Statement:
Using the Database explorer in DBeaver Version 21.3.3.202201221033 on both Windows and Linux Clients against a DB2 V11.5.6 database we have observed an inconsistent behavior. When opening some schema DBeaver runs the following slow query after displaying the table/view/etc options
[jcc][Time:2022-01-25-16:27:15.886][Thread:DBeaver: Read statistics for V2INAT2][PreparedStatement@74503a42]stmt_bidiTransform (SELECT
    TABNAME,
    SUM(DATA_OBJECT_P_SIZE + INDEX_OBJECT_P_SIZE + LONG_OBJECT_P_SIZE + LOB_OBJECT_P_SIZE + XML_OBJECT_P_SIZE) AS TOTAL_SIZE_IN_KB
FROM SYSIBMADM.ADMINTABINFO
WHERE TABSCHEMA=?
GROUP BY TABNAME) called
[jcc][Time:2022-01-25-16:27:15.886][Thread:DBeaver: Read statistics for V2INAT2][PreparedStatement@74503a42]stmt_bidiTransform not enabled (SELECT
    TABNAME,
    SUM(DATA_OBJECT_P_SIZE + INDEX_OBJECT_P_SIZE + LONG_OBJECT_P_SIZE + LOB_OBJECT_P_SIZE + XML_OBJECT_P_SIZE) AS TOTAL_SIZE_IN_KB
FROM SYSIBMADM.ADMINTABINFO
WHERE TABSCHEMA=?
GROUP BY TABNAME)
This stops you exploring any other schema/view//table until the query completes which can take 40+minutes in 120+ TByte database

At other times DBeaver does not run this query when you are exploring the database and you can explore multiple schema/table/view/etc with no delay

Questions
1) Why does DBeaver run this query? 
2) What criteria is used to decide it is necessary?
3) How can we prevent it running this query?

**Comment by LonwoLonwo** [comment]
Hello @robertwx 

This query collects tables statistics.

You can disable this process with the right click on the connection in the Database Navigator Tree -> Connection View -> "Show statistics info"

![2022-01-31 10_17_49-DBeaver Ultimate 21 3 3 - _SAMPLEDB_Script-117](https://user-images.githubusercontent.com/45152336/151753629-177d7d6f-5752-4640-bc73-8419a13e4ec0.png)

**Comment by robertwx** [comment]
That solves the problem

Thank you for your help
